lib/plugins/aws/provider/awsProvider.js

```diff
@@ -472,6 +472,21 @@ class AwsProvider {
     return this.serverless.service.provider.deploymentPrefix || 'serverless';
   }
 
+  getLogRetentionInDays() {
+    const rawRetentionInDays = this.serverless.service.provider.logRetentionInDays;
+    const retentionInDays = parseInt(rawRetentionInDays, 10);
+    if (_.isInteger(retentionInDays) && retentionInDays > 0) {
+      return retentionInDays;
+    }
+
+    const errorMessage = `logRetentionInDays should be an integer over 0 but ${rawRetentionInDays}`;
+    throw new this.serverless.classes.Error(errorMessage);
+  }
+
+  hasLogRetentionInDays() {
```

Comment:
Maybe we can have just `getLogRetentionInDays` which resolves with `null` in case of `logRetentionInDays` not being set.

At least having both `has` and `get` in this context seems to me unnecessarily verbose.

Reply:
Good point, no need to add an extra method just for the check in this case since we use the value everywhere we do the check anyways. I've removed hasLogRetentionInDays and made getLogRetentionInDays return null and fixed the associated checks.